### PR TITLE
[CameraRoll] Add assetThumbnail prop

### PR DESCRIPTION
**Pull request by Iragne**
Load Assets from Photo library and a memory warning is send.
Add a property in Image to force the display to use the [asset thumbnail] method

Related Issue https://github.com/facebook/react-native/issues/779

### CODE REVIEW

**Comment by Iragne** [comment]
any news ?

**Comment by bparadie** [comment]
@Iragne Your fix works beautifully for me!

**Comment by brentvatne** [comment]
ping @nicklockwood

File: Libraries/Image/Image.ios.js [hunk 1]
```diff
@@ -188,6 +197,7 @@ var nativeOnlyProps = {
   defaultImageSrc: true,
   imageTag: true,
   contentMode: true,
+  assetThumb:true,
```
Comment (bparadie) [inline Libraries/Image/Image.ios.js]: Hmm, shouldn't this be `assetThumbnail`?

**Comment by Iragne** [comment]
Yes agree with you 
Sorry for the error

**Comment by brentvatne** [comment]
@Iragne - can you rebase this and squash it into one commit? It looks like it won't be a clean merge

**Comment by bparadie** [comment]
@Iragne no worries. What puzzles me is that your code actually worked even with that bug...

**Comment by Iragne** [comment]
ok @brentvatne

**Comment by bparadie** [comment]
@Iragne I was curious whether you'll still get OOM crashes with thumbs and checked my camera roll with more than 200 videos. It ran fine without any problems. I believe for the thumbnail case you do not have to use a serial queue. In other words you could use this version (tested on my iPhone 6):

```
        if( !thumb ) {
          // ALAssetLibrary API is async and will be multi-threaded. Loading a few full
          // resolution images at once will spike the memory up to store the image data,
          // and might trigger memory warnings and/or OOM crashes.
          // To improve this, process the loaded asset in a serial queue.
          dispatch_async(RCTImageLoaderQueue(), ^{
            // Also make sure the image is released immediately after it's used so it
            // doesn't spike the memory up during the process.
            
            @autoreleasepool {
              UIImage *image = nil;
              ALAssetRepresentation *representation = [asset defaultRepresentation];
              ALAssetOrientation orientation = [representation orientation];
              image = [UIImage imageWithCGImage:[representation fullResolutionImage] scale:1.0f orientation:(UIImageOrientation)orientation];
              RCTDispatchCallbackOnMainQueue(callback, nil, image);
            }
          });
        } else {
          @autoreleasepool {
            UIImage *image = nil;
            ALAssetRepresentation *representation = [asset defaultRepresentation];
            ALAssetOrientation orientation = [representation orientation];
            image = [UIImage imageWithCGImage:[asset thumbnail] scale:1.0f orientation:(UIImageOrientation)orientation];
            RCTDispatchCallbackOnMainQueue(callback, nil, image);
          }
        }
```

@brentvatne Would you prefer to have that optimization as a separate PR?

**Comment by brentvatne** [comment]
@bparadie - yeah let's move that over to a separate PR! thanks

**Comment by Iragne** [comment]
Hello sorry complex weekend for me and i didn't have time to work on it.
I create a new PR Too mutch conflics with my version.
@bparadie I don't think it's good to not put the thumb in the queue you have a process to perform the thum creation. I correct the orrientation issue.

**Comment by bparadie** [comment]
@Iragne  no worries, it was just an experiment. The performance is pretty good even with the serial queue. @brentvatne suggested keeping the optimization separate. May I ask why you think it's not a good idea to use async thumb creation?

**Comment by tomstilljay1** [comment]
I have copyies

**Comment by Iragne** [comment]
@bparadie thanks if you have other point please let me know.
@brentvatne do you have time to validate it ?

**Comment by brentvatne** [comment]
Closing because of the new PR

**Pull request closed without merging**